modules/router/src/forwarding.ts

```diff
@@ -143,7 +147,8 @@ export async function forwardTransferCreation(
   // Below, we figure out the correct params needed for the receiver's channel. This includes
   // potential swaps/crosschain stuff
   let recipientAmount = senderAmount;
-  if (recipientAssetId !== senderAssetId) {
+  let recipientTransferDefinition = senderTransferDefinition;
+  if (recipientAssetId !== senderAssetId || recipientChainId !== senderChainId) {
```

Comment:
Why is there a chainId check here?

Reply:
Because cross-chain asset transfers still require swap profiles. Ex: does 1 mainnet ETH = 1 rinkeby ETH? If we didn't have this check, it would.